binarylog/binarylog_end2end_test.go

```diff
@@ -850,11 +850,25 @@ func equalLogEntry(entries ...*pb.GrpcLogEntry) (equal bool) {
 			tmp := append(h.Metadata.Entry[:0], h.Metadata.Entry...)
 			h.Metadata.Entry = tmp
 			sort.Slice(h.Metadata.Entry, func(i, j int) bool { return h.Metadata.Entry[i].Key < h.Metadata.Entry[j].Key })
+			// Delete headers that have POST values here since we cannot control
+			// this.
+			for i, entry := range h.Metadata.Entry {
+				if entry.Key == ":method" {
+					h.Metadata.Entry = append(h.Metadata.Entry[:i], h.Metadata.Entry[i+1:]...)
```

Comment:
This is wrong if there are more than one `:method` entry in the list. (It won't happen in practice though).

Either make it break after in the if.
Or `i++` only the key doesn't match `:method`.

Reply:
Hmmmm I see what you're saying, would delete from list as you're iterating? Broke after deleting :method.